crates/modelardb_manager/src/cluster.rs

```diff
@@ -66,23 +71,31 @@ impl Cluster {
     /// Remove the node with an url matching `url` from the current nodes, flush the node, and
     /// finally kill the process running on the node. If no node with `url` exists,
     /// [`ConfigurationError`](ModelarDbError::ConfigurationError) is returned.
-    pub async fn remove_node(&mut self, url: &str) -> Result<(), ModelarDbError> {
+    pub async fn remove_node(&mut self, url: &str, key: Uuid) -> Result<(), ModelarDbError> {
         if self.nodes.iter().any(|n| n.url == url) {
             self.nodes.retain(|n| n.url != url);
 
             // Flush the node and kill the process running on the node.
             let mut flight_client = FlightServiceClient::connect(format!("grpc://{url}"))
                 .await
-                .map_err(|error| ModelarDbError::ConfigurationError(error.to_string()))?;
+                .map_err(|error| ModelarDbError::ClusterError(error.to_string()))?;
 
             let action = Action {
                 r#type: "KillEdge".to_owned(),
                 body: vec![].into(),
             };
 
+            // Add the key to the request metadata to authenticate that the request is from the manager.
+            let mut request = Request::new(action);
+
+            // unwrap() is safe since a UUID cannot contain invalid characters.
+            request
+                .metadata_mut()
+                .insert("x-manager-key", key.to_string().parse().unwrap());
```

Comment:
This was the initial solution during development but i decided to use UUID in the context since I think it is preferable to always have the most specific type possible. String is a bit too broad and might even cause problems in some cases since there are limitations to what can be put in the tonic request metadata.

I have kept it as is for now. Cluster actions are fairly rare so there should not be any performance issues with this.